lib/aws/AwsSigv4Signer-sdk-v3.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+'use strict';
+const AwsSigv4SignerError = require('./errors');
+const { giveAwsV4Signer, loadAwsCredentialProvider } = require('./shared');
+
+const getAwsSDKCredentialsProvider = async () => {
+  try {
+    const awsV3 = await import('@aws-sdk/credential-provider-node');
+    if (typeof awsV3.defaultProvider === 'function') {
+      return awsV3.defaultProvider();
+    }
+  } catch (err) {
+    throw new AwsSigv4SignerError(
+      'Unable to find a valid AWS SDK, please provide a valid getCredentials function to AwsSigv4Signer options.'
```

Comment:
This error is difficult to action. I think a simple "Missing AWS SDK" error is fine. If you want to offer a hint, try something that one can actually do (e.g. add an import? a dependency?).